.github/list-prune-tags.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+function help () {
+    echo "List non-production git tags"
+    echo
+    echo "USAGE:"
+    echo "    list-prune-tags.sh [OPTIONS] [EXCLUDE_TAG ...]"
+    echo
+    echo "OPTIONS:"
+    echo "    -h           Display this help output and exit."
+    echo "    -n           Do not print a newline after each item."
+    echo
+    echo "PARAMETERS:"
+    echo "    EXCLUDE_TAG  (optional) Exclude this tag from the list of printed"
+    echo "                 tags. Multiple tags can be excluded."
+}
+
+function main () {
+    local -ar skip_tags=( $@ )
+    local -ar nonproduction_tags=( $(git tag -l | grep -v -E '^v[[:digit:]]{8}.[[:digit:]]{1,2}$') )
```

Comment:
This can be a follow-up since it shouldn't be an issue for a while and will complicate the current implementation, but we've started cutting tags for Azure CLI Preview extension releases, which follow the format `azext-aro-${API_VERSION}-${RELEASE_NUMBER}` e.g. `azext-aro-v20240812preview-00`. 

I'm fine however we choose to handle this, including changing the naming convention for the CLI extension tags (it's not set in stone anywhere other than documentation).

Reply:
I've included the CLI preview release tag in the list of tags to exclude from deletion.